c10/hip/CMakeLists.txt

```diff
@@ -48,9 +48,7 @@ if(NOT BUILD_LIBTORCHLESS)
   endif()
 
   # ---[ Dependency of c10_hip
-  target_link_libraries(c10_hip PUBLIC c10)
-
-  target_link_libraries(c10_hip PUBLIC ${PYTORCH_HIP_LIBRARIES})
+  target_link_libraries(c10_hip PUBLIC ${C10_LIB} hip::amdhip64)
```

Comment:
@naromero77amd Why the change from `c10` to `${C10_LIB}`?

Reply:
This is for consistency with what is done in [c10/cuda/CMakeLists.txt](https://github.com/pytorch/pytorch/blob/main/c10/cuda/CMakeLists.txt#L68)